authentication/securestore/main.go

```diff
@@ -116,13 +121,15 @@ func UpsertConfigToKeyring(backend string, creds []byte) error {
 }
 
 // RemoveConfigFromKeyring will remove the credentials from the first priority OS secure store.
-//
-// Note: CGO_ENABLED=1 is required for OSX Keychain and darwin builds
 func RemoveConfigFromKeyring(backend string) error {
 	if err := ValidateBackend(backend); err != nil {
 		return err
 	}
 
+	if runtime.GOOS == "darwin" && backend == "keychain" {
```

Comment:
nit: this check was used a couple of times, it probably deserves its own function